### DISCUSSION TITLE
Double Underline

### DISCUSSION THREAD

**Discussion by jck-incom** [QuestPDF/QuestPDF#534] [Ideas]
Just like the underline function, a way to make a double underline would be cool. 😄

It is often used for results in calculations and for extra emphasis in some invoices and such.

For example in some text editors you have underline, double underline, dotted underline, dashed underline, wavy underline and such.

I also just want to mention that I really find this library very promising, me and a friend are trying to look at many options for exporting documents to PDF, and this one is the most promising right now. Really great work on it! 🥇

**Comment by MarcinZiabek** [comment]
The single underline placement is usually defined by the font itself (both thickness and position of the line). I need to find out about a configuration for double underline. This case may be similar to the subscript and superscript, which are only simulated based on per-library convention. 

It is rather lower priority for now, but I am open for any PRs 😁

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for the kind words and the suggestion!

Great news — all the decoration styles you described are now available: `.DecorationDouble()`, `.DecorationDotted()`, `.DecorationDashed()`, `.DecorationWavy()`, and `.DecorationSolid()`. You can also customize decoration color and thickness. Full documentation here:
https://www.questpdf.com/api-reference/text/text-style.html

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
